TVnMoviesApp/Controller/Home/HomeViewController.swift

```diff
@@ -131,10 +164,10 @@ extension HomeViewController: SearchDelegate {
     func didUpdateMovie(_ data: SearchData) {
         if data.results!.count != 0 {
             for i in 0...data.results!.count - 1 {
-                resultsArray.append(data.results![i])
+                resultsArray.value.append(data.results![i])
```

Comment:
actually this entire method could just look like this 😄 
```swift
resultsArray.value = data.results ?? []
```
It's too much of a hassle with all these appends.
Other than that:
1. `!` - forced unwrap should be avoided. Some Swift developers even say you shouldn't use it at all. I don't agree with that though. I'm using it to express that I'm really sure that this optional should have value at this point and if it does't than it must be due to a programmer's error rather than any other random event.
2. If you're checking if an array is empty, a swift collection has `isEmpty` method, and it's a preferred, more modern way of checking such conditions. It just reads more nicely.
3. Much more common way of representing such for loop would be `for i in 0 ..< data.results!.count`...
4. ..but in this case you could just write `for result in data.results!`